### ISSUE TITLE
Single Layer Neural Network

### ISSUE THREAD

**Issue by irfanalidv** [irfanalidv/Applied_Deep_Neural_Networks_using_Keras#2]
<img width="945" alt="single_layer" src="https://cloud.githubusercontent.com/assets/5808185/26004183/956839c4-3752-11e7-9144-a8c4ae742c0f.png">

Weights Update Rule
We use a gradient descent optimization algorithm to learn the Weights Coefficients of the model. 

In every epoch (pass over the training set), we update the weight vector  ww  using the following update rule:
w=w+Δw,where Δw=−η∇J(w)
w=w+Δw,where Δw=−η∇J(w)
 

In other words, we computed the gradient based on the whole training set and updated the weights of the model by taking a step into the opposite direction of the gradient  ∇J(w)∇J(w) .
In order to fin the optimal weights of the model, we optimized an objective function (e.g. the Sum of Squared Errors (SSE)) cost function  J(w)J(w) .

Furthermore, we multiply the gradient by a factor, the learning rate  ηη  , which we choose carefully to balance the speed of learning against the risk of overshooting the global minimum of the cost function.
